### ISSUE TITLE
Build on iOS not working

### ISSUE THREAD

**Issue by ltropin** [rafaelsetragni/awesome_notifications#69]
When I try to run - "flutter run" on iOS Simulator
(Before doing this, I do the following: remove DerivedData, run "flutter clean", clean build folder on XCode.)
I get the following error:
`ld: warning: Could not find or use auto-linked framework 'Flutter'
    Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_FlutterStandardTypedData", referenced from:
          objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
      "_OBJC_CLASS_$_FlutterError", referenced from:
          objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
      "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
          objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)`

Your advice doesn't work:

> Issue: Undefined symbols for architecture arm64/armv7
> 
> Fix: XCode, to reduce the debug build time, only compiles for the current target architecure. For some reason, when you try to build on release and profile mode without clean your project, XCode tries to reuse those base libraries and fails because of their lack of other architectures.
> 
> Try to clean up your project in XCode (⇧ + ⌘ + K) and try to clean up your flutter project (flutter clean) to ensure that all debug files are not reusable in other enviroments.

Flutter doctor:

[✓] Flutter (Channel dev, 1.26.0-1.0.pre, on Mac OS X 10.15.7 19H2 darwin-x64, locale ru-RU)
    • Flutter version 1.26.0-1.0.pre at /Users/admin/development/flutter
    • Framework revision 63062a6443 (9 days ago), 2020-12-13 23:19:13 +0800
    • Engine revision 4797b06652
    • Dart version 2.12.0 (build 2.12.0-141.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/admin/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio 2.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0, Build version 12A7209
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio 2.app/Contents
    • Flutter plugin version 48.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.17.0

[✓] Connected device (2 available)
    • iPhone 11 (mobile) • BCDDA15A-4F07-4864-9FB6-C30606D5E81E • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 84.0.4147.135

**Comment by ahmetakil** [comment]
I'm also having the same issue. I get the following error in AwesomeServiceExtension and AwesomeContentExtension:

Could not find or use auto-linked framework 'Flutter'
Undefined symbol: _OBJC_CLASS_$_FlutterMethodChannel
Undefined symbol: _OBJC_CLASS_$_FlutterStandardTypedData
Undefined symbol: _OBJC_CLASS_$_FlutterError

**Comment by rafaelsetragni** [comment]
Please, check if this fix can help you:

[https://stackoverflow.com/questions/49767974/auto-linking-framework-not-found](https://stackoverflow.com/questions/49767974/auto-linking-framework-not-found)

Case it works, please tell me here and i gona improve the pluggin to automate this step.

**Comment by ahmetakil** [comment]
Hey @ltropin by any chance did you find a solution ?

**Comment by ltropin** [comment]
> Please, check if this fix can help you:
> 
> https://stackoverflow.com/questions/49767974/auto-linking-framework-not-found
> 
> Case it works, please tell me here and i gona improve the pluggin to automate this step.

I didn't really understand. I didn't find a solution ((

**Comment by ahmetakil** [comment]
I tried a lot of different solutions and unfortunately i wasn't able to find a solution as well

**Comment by rafaelsetragni** [comment]
I test my apps with the "flutter run --release", I don't know if there is a difference.

Im running and developing the iOS code using a virtual machine and is truly painful and slow. I gonna take some time to check it, unfortunately.

**Comment by ahmetakil** [comment]
If there is something i can do to help maybe debug certain parts please let me know

**Comment by thize** [comment]
0.0.5ˆ3 works, only 0.0.5ˆ4 that doesn't work, same issue here

**Comment by ahmetakil** [comment]
I tried downgrading to 0.0.5+3 but i still get the same error.

**Comment by thize** [comment]
try to click on import link "import 'package:awesome_notifications/awesome_notifications.dart';" and check folder version, 
![image](https://user-images.githubusercontent.com/10541221/103179223-6dbf8a00-4868-11eb-97bd-8619bc4e72a1.png) maybe the downgrade didn't work. I fixed it by placing the 'awesome_notifications: 0.0.5+3' on pubspec without 'ˆ'

**Comment by ahmetakil** [comment]
Yes i checked it the version is correct i tried 0.0.5+3,  0.0.5+2 and 0.0.5+1 unfortunately they all produce the same error

**Comment by ahmetakil** [comment]
As a test i cloned the example project in the repository, after  running pod install and launching i get the same error:

``ld: warning: Could not find or use auto-linked framework 'Flutter'
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_FlutterStandardTypedData", referenced from:
      objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
  "_OBJC_CLASS_$_FlutterError", referenced from:
      objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
  "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
      objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)``

**Comment by ahmetakil** [comment]
So @rafaelsetragni i generated verbose logs but i can't post it here it is too long, would it help if i sent it to you somehow through mail or something like discord ?

**Comment by hardik-weenggs** [comment]
did you try with different channel?
`flutter channel beta`

**Comment by ahmetakil** [comment]
Yes i tried with master and dev and my current "flutter --version" output is:
Flutter 1.26.0-1.0.pre • channel dev • https://github.com/flutter/flutter.git
Framework • revision 63062a6443 (2 weeks ago) • 2020-12-13 23:19:13 +0800
Engine • revision 4797b06652
Tools • Dart 2.12.0 (build 2.12.0-141.0.dev)

**Comment by rafaelsetragni** [comment]
Maybe is something related to pod files construction, but i do not discover the beat way yet.

**Comment by okursoftware** [comment]
We have the same problem, but i was able to debug some steps
1. Clone the github example project: Built fails 
2. Delete the ios folder of example project and run flutter create . : Build succeeds
3. Add the NotificationServiceExtension and complete the 9 steps: Build succeeds 

4. But if i ever run "flutter clean" command the app no longer compilers and i get the same error with everyone else  (Selecting Xcode's delete build folder option works fine the app still runs after that)

So i compared the program before 'flutter clean' and after 'flutter clean' there is almost no difference in ios folder only readable file that changes is /xcshareddata/scschemes/Runner.xscheme there are some changes

Also when i say Build succeeds i mean running 'flutter run' from terminal running it from Xcode always fails

ld: warning: Could not find or use auto-linked framework 'Flutter'
Undefined symbols for architecture x86_64:
"_OBJC_CLASS_$_FlutterStandardTypedData", referenced from:
objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
"_OBJC_CLASS_$_FlutterError", referenced from:
objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
"_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description

 flutter doctor -v
[✓] Flutter (Channel beta, 1.25.0-8.1.pre, on macOS 11.1 20C69 darwin-x64, locale tr-TR)
• Flutter version 1.25.0-8.1.pre at /Users/okur/Downloads/flutter
• Framework revision 8f89f6505b (2 weeks ago), 2020-12-15 15:07:52 -0800
• Engine revision 92ae191c17
• Dart version 2.12.0 (build 2.12.0-133.2.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
• Android SDK at /Users/okur/Library/Android/sdk
• Platform android-29, build-tools 29.0.3
• Java binary at: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/java
• Java version OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_275-b01)
• All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
• Xcode at /Applications/Xcode.app/Contents/Developer
• Xcode 12.3, Build version 12C33
• CocoaPods version 1.10.0

[!] Android Studio (not installed)
• Android Studio not found; download from https://developer.android.com/studio/index.html
(or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] Connected device (1 available)
• iPhone SE (2nd generation) (mobile) • 8E072B0C-EF11-4865-B13A-F245C15AE73A • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-3
(simulator)

**Comment by avenos47** [comment]
Hi is there a solution for this i can't publish my application because of this issue

**Comment by rafaelsetragni** [comment]
Yes, there is a solution. You help us to solve it instead of just complaining to the others about a free source code.

The problem is that we need to find a way to share the awesome_notifications library with the target extensions. The only way that i found to do it was to modify the .pod files.

If there us a better way to do it or a better modification on .pod files, im gonna be really glad to put your name on the project.

**Comment by nietsmmar** [comment]
@okursoftware  good work. Could you share what actually changed in `Runner.xscheme`?

**Comment by nietsmmar** [comment]
Has anyone else an idea of how this could be fixed? I'm totally in helping and trying to fix this. I just have no experience and no deeper knowledge with ios-native stuff.

**Comment by rafaelsetragni** [comment]
It is fixed on new 0.0.6+1 version. Please upgrade it and test it again.

It will be necessary do some changes in your source code. You can call me any time to help you to upgrade it.

**Comment by nietsmmar** [comment]
Awesome! I will test it tomorrow. Thank you for your work!

**Comment by nietsmmar** [comment]
I can't find how to get the `firebaseAppToken` and the `fcmTokenStream` in the new version. How is this done?

**Comment by nietsmmar** [comment]
I updated to the new version without using fcmtokens now. I used the same code like in the readme. But this is all I get when a notification comes in:
`D/FLTFireMsgReceiver( 9735): broadcast received for message`

`_firebaseMessagingBackgroundHandler` is never called.

**Comment by nietsmmar** [comment]
Ohh, I just realized that my app was still open and that I also need to implement:
`FirebaseMessaging.onMessage.listen((RemoteMessage message) {});` to react on foregroundmessages.

Now these callbacks are called. But `AwesomeNotifications().createNotificationFromJsonData(message.data);` won't work for me. It generates no notification.

**Comment by rafaelsetragni** [comment]
Can you print for us the content inside message.data?

**Comment by nietsmmar** [comment]
Of course @rafaelsetragni 

This is the `message.data` of the three send notifications:

```
{actionButtons: [{"key":"OPEN","label":"buttontext","autoCancel":true}], content: {"id":"11614010","channelKey":"daily_update","title":"","body":"this is some body","notificationLayout":"BigText","largeIcon":"somelink","backgroundColor":"0x00000000","groupKey":"daily_update_group","payload":{"id":"myId"}}}

{actionButtons: [{"key":"OPEN","label":"buttontext","autoCancel":true}], content: {"id":"11614110","channelKey":"daily_update","title":"","body":"this is some body","notificationLayout":"BigText","largeIcon":"somelink","backgroundColor":"0x00000000","groupKey":"daily_update_group","payload":{"id":"myId"}}}

{actionButtons: [{"key":"OPEN","label":"buttontext","autoCancel":true}], content: {"id":"11614210","channelKey":"daily_update","title":"","body":"this is some body","notificationLayout":"BigText","largeIcon":"somelink","backgroundColor":"0x00000000","groupKey":"daily_update_group","payload":{"id":"myId"}}}
```

This is how I initialized awesome_notifications:
```
AwesomeNotifications().initialize('resource://drawable/ic_notifications', [
    NotificationChannel(
        channelKey: 'daily_update',
        channelName: 'Daily interesting news channel',
        channelDescription: 'Notification channel for interesting daily news',
        defaultColor: Color(0xED5338),
        ledColor: Color(0xED5338),
        groupKey: "daily_update_group",
        groupAlertBehavior: GroupAlertBehavior.Children)
  ]);
```

**Comment by rafaelsetragni** [comment]
> This is the `message.data` of the three send notifications:
> 
> ```
> {actionButtons: [{"key":"OPEN","label":"buttontext","autoCancel":true}], content: {"id":"11614010","channelKey":"daily_update","title":"","body":"this is some body","notificationLayout":"BigText","largeIcon":"somelink","backgroundColor":"0x00000000","groupKey":"daily_update_group","payload":{"id":"myId"}}}
> 
> {actionButtons: [{"key":"OPEN","label":"buttontext","autoCancel":true}], content: {"id":"11614110","channelKey":"daily_update","title":"","body":"this is some body","notificationLayout":"BigText","largeIcon":"somelink","backgroundColor":"0x00000000","groupKey":"daily_update_group","payload":{"id":"myId"}}}
> 
> {actionButtons: [{"key":"OPEN","label":"buttontext","autoCancel":true}], content: {"id":"11614210","channelKey":"daily_update","title":"","body":"this is some body","notificationLayout":"BigText","largeIcon":"somelink","backgroundColor":"0x00000000","groupKey":"daily_update_group","payload":{"id":"myId"}}}
> ```

Id field must be different between the notifications, othewise one will be replaced by another. actionButtons and content must be a String object to jsonData be valid and backgroundColor must be "0", because 0x00000000 is how you describe an hexadecimal value in flutter. You must send a integer to set colors remotely:

 `0x00000000` is equivalent to  `0 (int)` or  `Colors.transparent`. 
 `0xff000000` is equivalent to   `4278190080 (int)` or `Colors.white`.
 `0xffffffff` is equivalent to   `4294967295 (int)` or `Colors.black`.

You can use [this](https://www.rapidtables.com/convert/number/hex-to-decimal.html) website to convert between hexadecimal and decimal values.

But i realy enjoyed how did you presented the color value! Way more didatic! I gonna implement this alternative pattern to represent colors in next release, for sure!

> This is how I initialized awesome_notifications:
> 
> ```
> AwesomeNotifications().initialize('resource://drawable/ic_notifications', [
>     NotificationChannel(
>         channelKey: 'daily_update',
>         channelName: 'Daily interesting news channel',
>         channelDescription: 'Notification channel for interesting daily news',
>         defaultColor: Color(0xED5338),
>         ledColor: Color(0xED5338),
>         groupKey: "daily_update_group",
>         groupAlertBehavior: GroupAlertBehavior.Children)
>   ]);
> ```

(Still testing it, wait just a little bit)

**Comment by nietsmmar** [comment]
I did use unique id's in all three notifications. I think the jsondata is correct as the notifications are shown. Only problem is, that they get replaced by each other as if they'd have the same id. Before the update I had a similar problem that I explained in this issue: https://github.com/rafaelsetragni/awesome_notifications/issues/146 The only difference is that back then 2/3 notifications were shown.

Oh now, I understand why setting colors never worked for me. Thanks!

**Comment by rafaelsetragni** [comment]
> I think the jsondata is correct

jsonDecode cannot convert into a valid map data if you arent  using 'actionButtons' and 'content' as String.

```dart
    String jsonData = r'{"actionButtons": [{"key":"OPEN","label":"buttontext","autoCancel":true}], '
        r'"content": {"id":"11614010","channelKey":"daily_update","title":"","body":"this is some body",'
        r'"notificationLayout":"BigText","largeIcon":"somelink","backgroundColor": 4278190080,'
        r'"groupKey":"daily_update_group","payload":{"id":"myId"}}}';

    Map<String, dynamic> mapData = jsonDecode(jsonData);
    print(mapData);
```

**Comment by nietsmmar** [comment]
Hm, I am a little bit confused as the notifications itself are shown. So the decode should work, or not? Also the Actionbutton is shown and is working.

I am even more confused now because this is how I send my notifications with java-admin-sdk: 
```
Message message = Message.builder()
                .setTopic(topic)
                .setAndroidConfig(AndroidConfig.builder()
                        .setPriority(AndroidConfig.Priority.HIGH)
                        .build())
                .setApnsConfig(ApnsConfig.builder()
                        .setAps(Aps.builder().setMutableContent(true).setContentAvailable(true).build())
                        .build())
                .putData("content", contentString)
                .putData("actionButtons", actionButtonsString)
                .setFcmOptions(FcmOptions.withAnalyticsLabel("SendToAll"))
                .build();
```
(this works, except grouping problem)

Now I changed it to this:
```
Message message = Message.builder()
                .setTopic(topic)
                .setAndroidConfig(AndroidConfig.builder()
                        .setPriority(AndroidConfig.Priority.HIGH)
                        .build())
                .setApnsConfig(ApnsConfig.builder()
                        .setAps(Aps.builder().setMutableContent(true).setContentAvailable(true).build())
                        .build())
                .putData("\"content\"", contentString)
                .putData("\"actionButtons\"", actionButtonsString)
                .setFcmOptions(FcmOptions.withAnalyticsLabel("SendToAll"))
                .build();
```

which leads to this `message.data`

`{"actionButtons": [{"key":"OPEN","label":"buttonLabel","autoCancel":true}], "content": {"id":"116222","channelKey":"daily_update","title":"","body":"some body text","notificationLayout":"BigText","largeIcon":"someLink","backgroundColor":"0x00000000","groupKey":"daily_update_group","payload":{"id":"myId"}}}`

and this won't work at all.

**Comment by rafaelsetragni** [comment]
About the first notification be missing, there is a thread on Stack Overflow about it [here](https://stackoverflow.com/questions/38097442/notification-with-setgroupsummarytrue-is-not-visible-in-android-n).

Im checking if there is a way to resolve this. I agree with you that this behaviour is not desired.

**Comment by rafaelsetragni** [comment]
I gonna release a new version soon with this Android standard behaviour fixed. Whe you receive the new version, test it for me:


```dart


  String jsonData1 = r'{ '
      r'"content": {"id":1,"channelKey":"daily_update","title":"a1","body":"this is some body 1",'
      r'"notificationLayout":"BigPicture",'
      r'"largeIcon":"https://media.fstatic.com/kdNpUx4VBicwDuRBnhBrNmVsaKU=/full-fit-in/290x478/media/artists/avatar/2013/08/neil-i-armstrong_a39978.jpeg",'
      r'"backgroundColor": 4278190080,'
      r'"groupKey":"daily_update_group","payload":{"id":"myId"}}}';

  String jsonData2 = r'{ '
      r'"content": {"id":2,"channelKey":"daily_update","title":"a2","body":"this is some body 2",'
      r'"notificationLayout":"BigPicture",'
      r'"largeIcon":"https://media.fstatic.com/kdNpUx4VBicwDuRBnhBrNmVsaKU=/full-fit-in/290x478/media/artists/avatar/2013/08/neil-i-armstrong_a39978.jpeg",'
      r'"backgroundColor": 4278190080,'
      r'"groupKey":"daily_update_group","payload":{"id":"myId"}}}';

  String jsonData3 = r'{ '
      r'"content": {"id":3,"channelKey":"daily_update","title":"a3","body":"this is some body 3",'
      r'"notificationLayout":"BigPicture",'
      r'"largeIcon":"https://media.fstatic.com/kdNpUx4VBicwDuRBnhBrNmVsaKU=/full-fit-in/290x478/media/artists/avatar/2013/08/neil-i-armstrong_a39978.jpeg",'
      r'"backgroundColor": 4278190080,'
      r'"groupKey":"daily_update_group","payload":{"id":"myId"}}}';

  String jsonData4 = r'{ '
      r'"content": {"id":4,"channelKey":"daily_update","title":"a4","body":"this is some body 4",'
      r'"notificationLayout":"BigPicture",'
      r'"largeIcon":"https://media.fstatic.com/kdNpUx4VBicwDuRBnhBrNmVsaKU=/full-fit-in/290x478/media/artists/avatar/2013/08/neil-i-armstrong_a39978.jpeg",'
      r'"backgroundColor": 4278190080,'
      r'"groupKey":"daily_update_group","payload":{"id":"myId"}}}';

  Map<String, dynamic> mapData1 = jsonDecode(jsonData1);
  Map<String, dynamic> mapData2 = jsonDecode(jsonData2);
  Map<String, dynamic> mapData3 = jsonDecode(jsonData3);
  Map<String, dynamic> mapData4 = jsonDecode(jsonData4);

  AwesomeNotifications().createNotificationFromJsonData(mapData1);
  sleep(Duration(seconds: 2));
  AwesomeNotifications().createNotificationFromJsonData(mapData2);
  sleep(Duration(seconds: 2));
  AwesomeNotifications().createNotificationFromJsonData(mapData3);
  sleep(Duration(seconds: 2));
  AwesomeNotifications().createNotificationFromJsonData(mapData4);
```

**Comment by rafaelsetragni** [comment]
I also added the hexadecimal representation color, as 0xff000000. Now this representation is a valid format for color values.

**Comment by nietsmmar** [comment]
I tested your code and  when I execute this after appstart, the grouping works for these notifications. But now..this is very very strange. If I send the exact same json with firebase and receiving it in  `FirebaseMessaging.onMessage.listen((RemoteMessage message)` and calling the `createNotificationFromJsonData` from there, grouping won't work. I made sure that it is the exact same data. I even printed it before calling the createNotification message and it is really the same.

Any idea on this? 

Also I finally tested on iOS. foregroundmessages worked for me. When the app was closed backgroundmessage weren't arriving. But this could also be some misconfiguration on my end..

**Comment by nietsmmar** [comment]
@rafaelsetragni I encountered another issue.

On Android the background messages worked but the foregroundmessage listener was never called although my log said:
`broadcast received for message`

I tried many thinks. Also because this was a known issue in FirebaseMessaging 9.0.0 see [here](https://stackoverflow.com/questions/66701257/flutter-2-0-with-firebase-cloud-messaging-onmessage-not-called-on-android).

But I realized that when I use `awesome_notifications 0.0.5+8` the listener is called. When I change it to `0.0.6+3` it is not called.

**Comment by nietsmmar** [comment]
Another think I noticed is, that colors are not working for me. I did not change my code but my Notification Icon and Actionbutton color is now different (gray). Before I had it in red. I Initialize it with `defaultColor: Color(0xED5338),`

**Comment by rafaelsetragni** [comment]
try `defaultColor: Color(0xffED5338)`

**Comment by nietsmmar** [comment]
@rafaelsetragni  I did set the color like this now but did not succeed.

**Comment by rafaelsetragni** [comment]
![image](https://user-images.githubusercontent.com/40064496/117144913-e4c1dd80-ad88-11eb-8ffd-ac56cc91e764.png)
It is working as expected. What platform are you using?

NOTE: Watch out! This color is not a web safe color.

**Comment by nietsmmar** [comment]
I reinstalled the app and also tried the `forceUpdate` but my result looks like this:

![2021-05-05_18-24](https://user-images.githubusercontent.com/9265181/117175768-7b18f200-adcf-11eb-91c5-ffac72db1d79.png)


It worked before the update though.

**Comment by rafaelsetragni** [comment]
This notification was created remotely? What android version are you using?

**Comment by nietsmmar** [comment]
Yes I created it with Firebase-Admin-SDK and send it with FCM.

I tested with Android 8.1.0 and Android 10

**Comment by rafaelsetragni** [comment]
Send me here the value of your message.data

**Comment by nietsmmar** [comment]
```
{"id":25,"channelKey":"daily_update","title":"hallo people?","body":"this is a test","notificationLayout":"Inbox","largeIcon":"","backgroundColor":"0x00000000","groupKey":"daily_update_group","payload":{"articleId":"JZ_1026920184"}}
[{"key":"OPEN","label":"Öffnen","autoCancel":true}]
```

**Comment by rafaelsetragni** [comment]
Try this one:

```Json
{"content":{"id":25,"channelKey":"daily_update","title":"hallo people?","body":"this is a test","notificationLayout":"Inbox","largeIcon":"","backgroundColor":"0xFFED5338","groupKey":"daily_update_group","payload":{"articleId":"JZ_1026920184"}},
"actionButtons":
[{"key":"OPEN","label":"Öffnen","autoCancel":true}]
}
```

**Comment by nietsmmar** [comment]
> Try this one:
> 
> ```json
> {"content":{"id":25,"channelKey":"daily_update","title":"hallo people?","body":"this is a test","notificationLayout":"Inbox","backgroundColor":"0xFFED5338","groupKey":"daily_update_group","payload":{"articleId":"JZ_1026920184"}},
> "actionButtons":
> [{"key":"OPEN","label":"Öffnen","autoCancel":true}]
> }
> ```

Oh wow, that worked. I just did not think that `backgroundColor` is actually the Color of the icon/title and buttontext.

**Comment by rafaelsetragni** [comment]
Yes. Right now bakcgroundColor name definetly does not translate their full behavior, but that is because i didnt finished this color programing. 

For the final release, the backgroundColor will work togheter with another primaryColor parameter, and in cases that this color could not be setted due platform limitations, it will change the behaviror, always privileging user legibility. Thats why backgroundColor today defines the icon and text colors for Android. This is the workaround behavior because i was not able to define the true background color, with the text in contrast.

**Comment by rafaelsetragni** [comment]
I gonna close this issue for now, since there is no more activity in 6 months. But you can reopen it, in case you judge necessary.
